XenAdmin/TabPages/HAPage.cs

```diff
@@ -197,6 +197,10 @@ private void Rebuild()
                 }
                 else
                 {
+                    //CA-250234 no need to rebuild HA page if we lost connection to the pool 
+                    if (pool.Connection == null)

```

Comment:
This is not enough here, as Connection can be non-null but acually not connected. I suggest:
'if (connection == null || connection.Session == null || !connection.IsConnected)'